File: ios/Classes/AdRequestManager.swift
// ios/Classes/AdRequestManager.swift

import Foundation

public class AdRequestManager {
  private var loadedInterstitialAds = [String: Bool]()
  private var loadedRewardedAds = [String: Bool]()

  public func loadInterstitialAd(placementId: String, callback: @escaping (Bool) -> Void) {
    // Simulate network delay
    DispatchQueue.global().asyncAfter(deadline: .now() + 1.0) {
      self.loadedInterstitialAds[placementId] = true
      callback(true)
    }
  }

  public func loadRewardedAd(placementId: String, callback: @escaping (Bool) -> Void) {
    DispatchQueue.global().asyncAfter(deadline: .now() + 1.0) {
      self.loadedRewardedAds[placementId] = true
      callback(true)
    }
  }

  public func isInterstitialAdLoaded(placementId: String) -> Bool {
    return loadedInterstitialAds[placementId] ?? false
  }

  public func isRewardedAdLoaded(placementId: String) -> Bool {
    return loadedRewardedAds[placementId] ?? false
  }
}

File: android/src/main/kotlin/com/example/ad_platform_sdk/AdRequestManager.kt
package com.example.ad_platform_sdk

class AdRequestManager {

    // Simple in-memory cache for loaded ads by their placement ID
    private val loadedInterstitialAds = mutableMapOf<String, Boolean>()
    private val loadedRewardedAds = mutableMapOf<String, Boolean>()

    // Mock loading interstitial ad
    fun loadInterstitialAd(placementId: String, callback: (success: Boolean) -> Unit) {
        // Simulate network/load delay
        Thread {
            Thread.sleep(1000)
            loadedInterstitialAds[placementId] = true
            callback(true)
        }.start()
    }

    // Mock loading rewarded ad
    fun loadRewardedAd(placementId: String, callback: (success: Boolean) -> Unit) {
        Thread {
            Thread.sleep(1000)
            loadedRewardedAds[placementId] = true
            callback(true)
        }.start()
    }

    fun isInterstitialAdLoaded(placementId: String): Boolean {
        return loadedInterstitialAds[placementId] ?: false
    }

    fun isRewardedAdLoaded(placementId: String): Boolean {
        return loadedRewardedAds[placementId] ?: false
    }
}
